stories/module-adsense-components.stories.js

```diff
@@ -411,6 +412,266 @@ generateReportBasedWidgetStories( {
+								'22',
+							],
+						},
+					],
+				},
+				{
+					dimensions: [
+						'Site Kit Top Earning Page 6',
+						'/site-kit-top-earning-page-6/',
+					],
+					metrics: [
+						{
+							values: [
+								'0.036977',
+								'0.33015178571428566',
+								'144',
+							],
+						},
+					],
+				},
+				{
+					dimensions: [
+						'Site Kit Top Earning Page 7',
+						'/site-kit-top-earning-page-7/',
+					],
+					metrics: [
+						{
+							values: [
+								'0.029555',
+								'0.29555',
+								'206',
+							],
+						},
+					],
+				},
+				{
+					dimensions: [
+						'Site Kit Top Earning Page 8',
+						'/site-kit-top-earning-page-8/',
+					],
+					metrics: [
+						{
+							values: [
+								'0.028485',
+								'1.0173214285714285',
+								'35',
+							],
+						},
+					],
+				},
+				{
+					dimensions: [
+						'Site Kit Top Earning Page 9',
+						'/site-kit-top-earning-page-9/',
+					],
+					metrics: [
+						{
+							values: [
+								'0.024269',
+								'0.3677121212121212',
+								'81',
+							],
+						},
+					],
+				},
+				{
+					dimensions: [
+						'Site Kit Top Earning Page 10',
+						'/site-kit-top-earning-page-10/',
+					],
+					metrics: [
+						{
+							values: [
+								'0.019556',
+								'1.777818181818182',
+								'13',
+							],
+						},
+					],
+				},
+			],
+			totals: [
+				{
+					values: [
+						'2.150211',
+						'0.6847805732484076',
+						'4304',
+					],
+				},
+			],
+			minimums: [
+				{
+					values: [
+						'0.0',
+						'0.0',
+						'1',
+					],
+				},
+			],
+			maximums: [
+				{
+					values: [
+						'0.76352',
+						'15.2164',
+						'825',
+					],
+				},
+			],
+		},
+	},
+];
+
+const optionsModuleTopEarningPagesWidget = {
+	startDate: '2020-08-15',
+	endDate: '2020-09-11',
+	dimensions: [ 'ga:pageTitle', 'ga:pagePath' ],
+	metrics: [
+		{ expression: 'ga:adsenseRevenue', alias: 'Earnings' },
+		{ expression: 'ga:adsenseECPM', alias: 'Page RPM' },
+		{ expression: 'ga:adsensePageImpressions', alias: 'Impressions' },
+	],
+	orderby: {
+		fieldName: 'ga:adsenseRevenue',
+		sortOrder: 'DESCENDING',
+	},
+	limit: 10,
+};
+
+generateReportBasedWidgetStories( {
+	moduleSlugs: [ 'adsense', 'analytics' ],
+	datastore: MODULES_ANALYTICS,
+	group: 'AdSense Module/Components/Module/Top Earning Pages Widget',
+	referenceDate: '2020-09-12',
+	data: dataModuleTopEarningPagesWidget,
+	options: optionsModuleTopEarningPagesWidget,
+	Component: ModuleTopEarningPagesWidget,
+	wrapWidget: false,
+	additionalVariants: {
+		'AdSense Not Linked': {
+			data: [],
+			options: optionsModuleTopEarningPagesWidget,
+		},
+		Loading: {
+			data: dataModuleTopEarningPagesWidget,
+			options: optionsModuleTopEarningPagesWidget,
+		},
+	},
+	additionalVariantCallbacks: {
+		Loading: ( dispatch, data, options ) => {
+			dispatch( MODULES_ANALYTICS ).setAdsenseLinked( true );
+			dispatch( MODULES_ANALYTICS ).startResolution( 'getReport', [ options ] );
+		},
```

Comment:
We don't need to reset the resolution status anymore, it's done in the generator function:

```suggestion
		Loading: ( dispatch ) => dispatch( MODULES_ANALYTICS ).setAdsenseLinked( true ),
```